src/cargo/sources/registry/http_remote.rs

```diff
@@ -438,6 +438,13 @@ impl<'cfg> RegistryData for HttpRegistry<'cfg> {
             return Poll::Ready(Ok(LoadResponse::NotFound));
         }
 
+        if self.config.offline() || self.config.cli_unstable().no_index_update {
```

Comment:
Should we add a test for the `self.config.cli_unstable().no_index_update` condition?

Reply:
Sure.